test-apps/display-test-app/src/frontend/CuratedCesiumContentProvider.ts

```diff
@@ -0,0 +1,154 @@
+import { CesiumTerrainAssetId } from "@itwin/core-common";
```

Comment:
Are the contents of this file intended to represent some public open-source package we will make available for any app to use?

Reply:
Ideally not, it would be sample code in our test apps and maybe a sample in the sandbox, as well as our docs, of how app teams can implement this themselves

Reply:
You expect every app to have to implement / copy-paste this big ugly blob of code just to be able to use the terrain that they currently get just by providing an API key? That seems like a huge downgrade and quite error-prone.

Reply:
Granted its very ugly right now, and needs to be cleaned up, but there isn't that much code that needs to be added.  App developers can customize it further to meet their needs, eg not rely on IModelApp to obtain access token, they might use an http client that additional caching, etc. Introducing another package to our already growing number of packages, doesn't seem responsible, especially as this is a beta api and very likely to change

Reply:
You have effectively defined a client library for the new curated content API already. Do you really expect every consumer to write their own interfaces and enums representing types *defined by that REST API*? Write their own logic to rustle up an iTwin Id to pass to an API that does not give two hoots what iTwin Id you give it?

I understand not wanting to have a direct call to iTP from within itwinjs-core. But can we show a little regard for our users who will have to deal with all of this unnecessary new friction? Where's the value-add to justify it?

Reply:
> it should be as simple

Can you please make it look simple then?